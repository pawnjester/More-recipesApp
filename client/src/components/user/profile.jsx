import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import NavigationBar from '../NavigationBar';
import '../../styles/profile.scss';
import getUserDetail from '../../actions/getUserDetail';
import EditUserModal from '../Modal/EditUserModal';
/* eslint-disable */
class Profile extends Component {
  constructor(props) {
    super(props);

    this.state = {
      modal: false,
    };
    this.toggle = this.toggle.bind(this);
  }

  componentWillMount() {
    // TODO: get recipes list
    this.props.getUserDetail();
  }

  toggle() {
    this.setState({
      modal: !this.state.modal,
    });
  }

  render() {
    const detail = (this.props.userDetail) ? this.props.userDetail : {};
    return (
      <div>
        <NavigationBar />
        <div className="heading mt-5">
          <div className="dark-overlay">
            <div className="container ">
              <div className="row profile">
                <div className="col-md-3 col-sm-12 col-xs-12">
                  <div className="profile-sidebar">
                    <div className="profile-user-pic">
                      <img src={detail.profileImg || '/food1.jpg'} alt="" className="img-fluid img-circle mx-auto d-block" />
                    </div>
                    <div className="profile-user-title">
                      <div className="profile-user-name">
                        {detail.username}
                      </div>
                      <div className="profile-user-job">
                        {detail.email}
                      </div>
                    </div>
                    <div className="profile-user-buttons">
                      <button className="btn btn-danger btn-md" onClick={this.toggle} styles="cursor:pointer"> <i className="fa fa-pencil" aria-hidden="true" /> Edit</button>
                      <EditUserModal
                        editUser={detail} // state
                        isOpen={this.state.modal}
                        toggle={this.toggle}
                        getuserDetail={this.props.getUserDetail}
                      />
                    </div>
                    <div className="profile-user-menu">
                      <ul className="list-group">
                        <li className=" list-group-item activ change"><a href=""><i className="fa fa-home" aria-hidden="true" />    Overview</a></li>
                        <li className="list-group-item change" ><Link to="/change-password"><i className="fa fa-key" aria-hidden="true">  Change Password</i></Link></li>
                      </ul >
                    </div>
                  </div>
                </div>
                <div className="col-md-9">
                  <div className="card" >
                    <div className="card-body">
                      <table className="table table-borderless mt-3">
                        <tbody>
                          <tr>
                            <td className="text-dark"><b> Name</b></td>
                            <td className="text-dark">{detail.username}</td>
                          </tr>
                          <tr>
                            <td className="text-dark"><b>Phone Number</b></td>
                            <td>{}</td>
                          </tr>
                          <tr>
                            <td className="text-dark"><b>Fax</b></td>
                            <td>{}</td>
                          </tr>
                          <tr>
                            <td className="text-dark"><b>Email</b></td>
                            <td className="text-dark">{detail.email}</td>
                          </tr>
                          <tr>
                            <td className="text-dark"><b>Address</b></td>
                            <td>{}</td>
                          </tr>
                          <tr>
                            <td className="text-dark"><b>Country</b></td>
                            <td>{}</td>
                          </tr>
                          <tr>
                            <td className="text-dark"><b>Marital Status</b></td>
                            <td>{}</td>
                          </tr>
                        </tbody>
                      </table>
                    </div>
                  </div>
                </div>
              </div>

            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  userDetail: state.userDetailReducer.userDetail,
});


export default connect(
  mapStateToProps,
  {
    getUserDetail,
  },
)(Profile);
