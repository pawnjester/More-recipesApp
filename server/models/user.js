/** User database model/association
 * @exports User
 * @param  {object} sequelize - sequelize
 * @param  {object} DataTypes - sequelize Datatypes
 * @return {object} The User model
 */
export default (sequelize, DataTypes) => {
  const User = sequelize.define('User', {
    username: {
      type: DataTypes.STRING,
      allowNull: false,
      unique: true
    },
    email: {
      type: DataTypes.STRING,
      validate: {
        isEmail: {
          msg: 'Invalid email'
        },
      }
    },
    password: {
      type: DataTypes.STRING,
      allowNull: false,
    }
  });

   //salt and hash passwords before creating users 
  User.beforeCreate((user, options) => {
  const salt = bcrypt.genSaltSync();
  user.password = bcrypt.hashSync(user.password, salt);
});

  //compare the hash password to the password given
  User.prototype.validPassword = function(password) {
    return bcrypt.compareSync(password, this.password);
  };

  //Instance method to generate token for the user
  User.prototype.generateAuthToken = function generateAuthToken() {
    const user = this;
    const access = 'auth';
    const token = jwt.sign({id: user.id, access},
     process.env.SECRET_KEY,
     { expiresIn: Math.floor(new Date().getTime()/1000) + 7*24*60*60 }).toString();    
    return token;

  }

  // Instance method to prevent password from
  // being sent to client.
  User.prototype.toJSON = function toJSON() {
    const values = Object.assign({}, this.get());

    delete values.password;
    return values;
  };
  User.associate = (models) => {
    User.hasMany(models.Recipe, { foreignKey: 'userId' });
  };

  return User;
};
