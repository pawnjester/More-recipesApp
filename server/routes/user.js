import express from 'express';
import User from '../controllers/users';
import authenticate from '../middleware/authenticate';



const userController = new User();

const router = express.Router();


router.post('/signup', userController.signUp);
router.post('/signin', userController.signIn);
router.get('/me', authenticate, userController.me);


export default router;
