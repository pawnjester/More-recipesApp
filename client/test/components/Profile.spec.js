import React from 'React';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect';
import { Profile } from '../../src/components/user/Profile';
import store from '../../src/store/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import sinon from 'sinon';


configure({ adapter: new Adapter() });

const props = {
  getUserDetail: jest.fn(),
  userDetail: {
    id: 1,
    username: 'charles',
    email: 'charles@example.com'
  },
  errors: {}
};

const setup = () => {
  const shallowWrapper = shallow(<Profile {...props} />);
  return {
    shallowWrapper,
  };
};

describe('User Profile Component', () => {
  it('should render correctly', () => {
    const { shallowWrapper } = setup();
    expect(shallowWrapper).toMatchSnapshot();
  });
});
