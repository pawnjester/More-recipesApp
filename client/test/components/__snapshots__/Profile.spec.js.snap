// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Profile Component should render correctly 1`] = `
<div>
  <Connect(NavigationBar) />
  <div
    className="profile-header"
  >
    <div
      className="dark"
    >
      <div
        className="container "
      >
        <div
          className="row profile-page"
        >
          <div
            className="col-md-5 col-sm-12 col-xs-12 mobile"
          >
            <div
              className="profile-sidebar"
            >
              <div
                className="profile-user-pic"
              >
                <img
                  alt=""
                  className="img-fluid img-circle mx-auto d-block"
                  src="/food1.jpg"
                />
              </div>
              <div
                className="profile-user-title"
              >
                <div
                  className="profile-user-name"
                >
                  charles
                </div>
                <div
                  className="profile-user-job"
                >
                  charles@example.com
                </div>
              </div>
              <div
                className="profile-user-menu"
              >
                <ul
                  className="list-group"
                >
                  <li
                    className=" list-group-item activ change"
                  >
                    <a
                      href=""
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-home"
                      />
                      Overview
                    </a>
                  </li>
                  <li
                    className="list-group-item change"
                  >
                    <Link
                      className="change-password-link"
                      replace={false}
                      to="/change-password"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-key"
                      >
                        Change Password
                      </i>
                    </Link>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            className="col-md-7 col-sm-12 col-xs-12 mobile"
          >
            <div
              className="card card-mix"
            >
              <div
                className="card-body"
              >
                <table
                  className="table table-borderless mt-3"
                >
                  <tbody>
                    <tr>
                      <td
                        className="text-dark"
                      >
                        <b>
                           Username
                        </b>
                      </td>
                      <td
                        className="text-dark"
                      >
                        charles
                      </td>
                    </tr>
                    <tr>
                      <td
                        className="text-dark"
                      >
                        <b>
                          Email
                        </b>
                      </td>
                      <td
                        className="text-dark"
                      >
                        charles@example.com
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
